### PR TITLE
VideoPress: Input and SearchInput components

### LINKED ISSUE #25749: VideoPress components: SearchInput
Image

### PR DESCRIPTION
**Pull request by dhasilva**
Closes #25749

#### Changes proposed in this Pull Request:

* Add `Input` and `SearchInput` components to VideoPress Package
* Fix some typescript issues

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all stories of `Packages/VideoPress/Input` component works as expected

Screenshot_2022-08-31_15-21-25
Screenshot_2022-08-31_15-20-54
Screenshot_2022-08-31_15-22-16
Screenshot_2022-08-31_15-22-05
Screenshot_2022-08-31_17-24-27

### CODE REVIEW

File: projects/packages/videopress/src/client/admin/components/input/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import { InputProps, SearchInputProps } from './types';
+import type React from 'react';
+
+/**
+ * Input component
+ *
+ * @param {InputProps} props - Component props.
+ * @returns {React.ReactNode} - Input react component.
+ */
+export const Input: React.FC< InputProps > = ( {
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/input/index.tsx]: We should discourage the usage of `React.FC`.

Context:

https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components
https://github.com/facebook/create-react-app/pull/8177

File: projects/packages/videopress/src/client/admin/components/input/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	const handleChangeEvent = useCallback(
+		( e: ChangeEvent< HTMLInputElement > ) => {
+			if ( onChange != null ) {
+				onChange( e.currentTarget.value );
+			}
+		},
+		[ onChange ]
+	);
+
+	const handleKeyDownEvent = useCallback(
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/input/index.tsx]: I prefer to do the action at `keyUp`, to make it the last action, but maybe is just a personal preference.

File: projects/packages/videopress/src/client/admin/components/input/index.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		( e: KeyboardEvent< HTMLInputElement > ) => {
+			if ( onEnter != null && e.key === 'Enter' ) {
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/input/index.tsx]:
```suggestion
                        if ( onEnter != null && e.code === 'Enter' ) {
```

Better to use the one provided by React
https://blog.saeloun.com/2021/04/23/react-keyboard-event-code.html

File: projects/packages/videopress/src/client/admin/components/input/index.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+				onEnter( e.currentTarget.value );
+			}
+		},
+		[ onEnter ]
+	);
+
+	return (
+		<div className={ classnames( className, styles.wrapper ) }>
+			{ hasIcon && icon }
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/input/index.tsx]:
```suggestion
                        { icon }
```

If `icon` is `undefined` or `null`, if you render it, that's no problem, React just ignore it.

File: projects/packages/videopress/src/client/admin/components/input/index.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+export const SearchInput: React.FC< SearchInputProps > = ( {
+	placeholder = __( 'Search your library', 'jetpack-videopress-pkg' ),
+	...componentProps
+} ) => {
+	return (
+		<Input
+			{ ...componentProps }
+			icon={ <SearchIcon size={ 24 } /> }
+			placeholder={ placeholder }
+		></Input>
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/input/index.tsx]:
```suggestion
                />
```

File: projects/packages/videopress/src/client/admin/components/input/style.module.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+.wrapper {
+	width: 100%;
+	height: calc( var( --spacing-base ) * 5 ); // 40px;
+	padding: var( --spacing-base );
+	border: 1px solid var( --jp-black );
+	border-radius: 4px;
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/input/style.module.scss]:
```suggestion
        border-radius: var(--jp-border-radius);
```

You can find all the available vars here: https://github.com/Automattic/jetpack/blob/trunk/projects/js-packages/components/components/theme-provider/index.tsx

File: projects/packages/videopress/src/client/admin/components/input/style.module.scss [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+	background-color: var( --jp-white );
+	display: flex;
+
+	&:focus-within {
+		border-width: 2px;
+		padding: calc( var( --spacing-base ) - 1px ); // 7px;
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/input/style.module.scss]: We shouldn't do this. Why exactly?

**Review by renatoagds — CHANGES_REQUESTED** [review]

File: projects/packages/videopress/src/client/admin/components/input/style.module.scss (hunk 7 above)
Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/input/style.module.scss]: To keep the input content static while increasing the border width.
I checked here and `InputControl` uses `box-shadow` instead. I'll change it.

File: projects/packages/videopress/src/client/admin/components/input/index.tsx (hunk 2 above)
Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/input/index.tsx]: Makes sense!

File: projects/packages/videopress/src/client/admin/components/input/style.module.scss (hunk 7 above)
Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/input/style.module.scss]: Done.

**Review by renatoagds — APPROVED** [review]
LGTM

**Pull request merged**